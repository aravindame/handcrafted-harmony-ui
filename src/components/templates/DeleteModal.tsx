import Modal from '@/components/molecules/Modal'

interface DeleteModalProps {
  isModalVisible: boolean
  onConfirm: () => void
  onCancel: () => void
}

const DeleteModal = ({onConfirm, onCancel, isModalVisible}: DeleteModalProps) =>
  <Modal
    isModalVisible={isModalVisible}
    title="Confirm Delete"
    message="Are you sure you want to delete this craft?"
    confirmBtnText="Delete"
    variant="danger"
    onConfirm={onConfirm}
    onCancel={onCancel}
  />

export default DeleteModal
