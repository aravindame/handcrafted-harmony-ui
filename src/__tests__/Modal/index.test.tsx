import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import DeleteModal from '@/components/templates/DeleteModal'

describe('Modal', () => {
  it('renders Delete modal', () => {
    const component = render(
      <DeleteModal
        isModalVisible={true}
        onConfirm= {() => {return}}
        onCancel= {() => {return}}
      />
    )

    expect(component).toMatchSnapshot()
  })
})
