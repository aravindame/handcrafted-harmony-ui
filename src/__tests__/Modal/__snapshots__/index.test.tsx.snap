// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders Delete modal 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    data-rr-ui-modal-open=""
    style="padding-right: 1024px;"
  >
    <div />
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-modal="true"
      class="fade modal show"
      role="dialog"
      style="display: block; padding-right: 0px;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <div
              class="modal-title h4"
            >
              Confirm Delete
            </div>
            <button
              aria-label="Close"
              class="btn-close"
              type="button"
            />
          </div>
          <div
            class="modal-body"
          >
            Are you sure you want to delete this craft?
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              Cancel
            </button>
            <button
              class="btn btn-danger"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
